aten/src/ATen/native/cpu/HistogramKernel.cpp

```diff
@@ -0,0 +1,209 @@
+#include <ATen/native/Histogram.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/native/BucketizationUtils.h>
+
+#include <algorithm>
+#include <mutex>
+#include <tuple>
+
+namespace at { namespace native {
+
+namespace {
+
+const int64_t HISTOGRAM_GRAIN_SIZE = 200;
+
+template<typename input_t, typename output_t, typename ExtractWeight_t, bool LinearBinEdges>
+void histogram_cpu_contiguous(Tensor& hist, const Tensor& bin_edges, const Tensor& input, const Scalar& min, const Scalar& max, ExtractWeight_t extract_weight) {
+    int64_t numel_in = input.numel();
```

Comment:
It may be appropriate to add some TORCH_INTERNAL_ASSERTs here for properties this function relies on, like contiguity